src/Entity/Delivery.php

```diff
@@ -298,6 +288,25 @@ public static function createWithTasks(Task ...$tasks)
         return $delivery;
     }
 
+    public function withTasks(Task ...$tasks)
+    {
+        $this->items->clear();
+
+        $pickup = array_shift($tasks);
+
+        // Make sure the first task is a pickup
+        $pickup->setType(Task::TYPE_PICKUP);
+
+        $this->addTask($pickup);
+
+        foreach ($tasks as $dropoff) {
+            $dropoff->setPrevious($pickup);
```

Comment:
How does it work this `previous` concept? 
If I have 3 tasks (1 pickup and 2 dropoffs) in my mind I think that:
- first task is pickup, doesn't have a previous
- second task, its previous is the pickup task and the next the other dropoff
- third task, its previous is the second dropoff and doesn't have a next

but here for both dropdoff tasks the previous it's always the pickup :thinking:

Reply:
In the case of a delivery with multiple tasks (a pickup + 2 dropoffs like you say), we don't necessarily know the order of dropofs. It's because we can't trust clients/users to be sending data that is already optimised. 

When completing a task, there is a piece code that checks if [the previous task has been completed](https://github.com/coopcycle/coopcycle-web/blob/315f848a4607a091d237390e0cc87c2340f1f2c6/src/Domain/Task/Handler/MarkAsDoneHandler.php#L36-L38). The pickup will always need to be completed first, but the dropoffs can be completed in any order.